### PR TITLE
ENH: copy_to supports aio files

### PR DESCRIPTION
**Pull request by ChengjieLi28**
## Check code requirements

- [x] tests added / passed (if needed)
- [x] Ensure all linting tests pass

### CODE REVIEW

File: python/xoscar/context.pyx [hunk 1]
```diff
@@ -294,12 +325,31 @@ cdef class ClientActorContext(BaseActorContext):
...
         context = self._get_backend_context(address)
-        return context.copy_to(local_buffers, remote_buffer_refs, block_size)
+        return context.file_object_ref(address, file_object)
+
+    def copy_to(self, local_buffers_or_fileobjs: list, remote_refs: List[Union[BufferRef, FileObjectRef]], block_size: Optional[int] = None):
+        if len(local_buffers_or_fileobjs) == 0 or len(remote_refs) == 0:
+            raise ValueError("Nothing to transfer since the length of `local_buffers_or_fileobjs` or `remote_refs` is 0.")
+        assert (
+            len({ref.address for ref in remote_refs}) == 1
+        ), "remote_refs for `copy_to` can support only 1 destination"
```
Comment (codingl2k1) [inline python/xoscar/context.pyx:337]: Why define the parameter `remote_refs` as list but check the len of value == 1?

File: python/xoscar/core.pyx [hunk 2]
```diff
@@ -586,3 +587,41 @@ cdef class BufferRef:
...
+        self.address = address
+
+    @classmethod
+    def create(cls, fileobj: AioFileObject, address: str, uid: bytes) -> "FileObjectRef":
+        ref = FileObjectRef(address, uid)
+        cls._ref_to_fileobjs[ref] = fileobj
+        return ref
+
+    @classmethod
+    def get_file_object(cls, ref: "FileObjectRef") -> AioFileObject:
```
Comment (codingl2k1) [inline python/xoscar/core.pyx:609]:
```suggestion
    def get_local_file_object(cls, ref: "FileObjectRef") -> AioFileObject:
```

File: python/xoscar/core.pyx [hunk 3]
```diff
@@ -586,3 +587,41 @@ cdef class BufferRef:
...
+    def __hash__(self):
+        return hash((self.address, self.uid))
+
+    def __eq__(self, other):
+        if type(other) != FileObjectRef:
+            return False
+        return self.address == other.address and self.uid == other.uid
+
+    def __repr__(self):
+        return f'FileObjectRef(uid={self.uid}, address={self.address}'
```
Comment (codingl2k1) [inline python/xoscar/core.pyx:627]:
```suggestion
        return f'FileObjectRef(uid={self.uid}, address={self.address})'
```

Comment (codingl2k1) [reply python/xoscar/core.pyx:627]: Printing hex of uid is helpful for debugging.

**Review by codingl2k1 — COMMENTED** [review]
Maybe we can use `sendfile` to optimize performance on the Linux platform and add an asv benchmark.

File: python/xoscar/context.pyx (hunk 1 above)
Comment (ChengjieLi28) [reply python/xoscar/context.pyx:337]: > Why define the parameter `remote_refs` as list but check the len of value == 1?

Here's checking whether there is only on remote address, i.e. the length of ``set`` of remote addresses is 1.

**Comment by ChengjieLi28** [comment]
> Maybe we can use `sendfile` to optimize performance on the Linux platform and add an asv benchmark.

Sure. Here's an issue #34 . Will have another PR later.

File: python/xoscar/context.pyx (hunk 1 above)
Comment (codingl2k1) [reply python/xoscar/context.pyx:337]: I got it. The copy_to API is too complex.

File: python/xoscar/core.pyx (hunk 3 above)
Comment (ChengjieLi28) [reply python/xoscar/core.pyx]: Fix.

File: python/xoscar/core.pyx (hunk 2 above)
Comment (ChengjieLi28) [reply python/xoscar/core.pyx]: Fix. Other places need call this func. So I just fix this in a new commit.

**Review by codingl2k1 — APPROVED** [review]
LGTM

**Review by aresnow1 — APPROVED** [review]
LGTM

**Pull request merged**